browser/brave_vpn/brave_vpn_service_factory.cc

```diff
@@ -111,6 +111,7 @@ BraveVpnService* BraveVpnServiceFactory::GetForProfile(Profile* profile) {
 void BraveVpnServiceFactory::BindForContext(
     content::BrowserContext* context,
     mojo::PendingReceiver<brave_vpn::mojom::ServiceHandler> receiver) {
+  CHECK(GetInstance());
```

Comment:
this doesn't look right. `BraveVpnServiceFactory::GetInstance()` should never return `nullptr`.

https://github.com/brave/brave-core/pull/20500 should be reimplemented in a different way. All checks should be done on `service` returned by a factory, not the factory itself.

`IsBraveVPNWireguardEnabled()` should be rewritten as:
```
bool IsBraveVPNWireguardEnabled(PrefService* local_state) {
  if (!IsBraveVPNFeatureEnabled()) {
    return false;
  }
  ...
```

DCHECKs in:
```
BraveVpnWireguardObserverFactory::GetServiceForContext()
BraveVpnDnsObserverFactory::GetServiceForContext()
```
should also be converted into conditions.

Reply:
Thanks! updated.